### PR TITLE
Move to C#'s lightweight property syntax.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This updates the LS to move all properties of the form:

```
Type Property
{
    get
    {
        return simpleExpression;
    }
}
```

Into the form: ```Type Property => simpleExpression;```.

Certain things had to be true to make this change.

1. The accessor list could not contain any form of trivia in it (beyond whitespace).  One exception to this is trivia on the semicolon of the return statement.  Because we use this same semicolon in the final property declaration, it's fine for it to have trivia.
2. The accessor could not have attributes on it.
3. The simple expression could only be an expression comprised of this/base/identifiers/dots.

In other words, these were actually really simple properties that read far better in their new form.   I did not change any properties that were more complex as i wasn't sure if the new form would be desirable.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide  This deletes 3300 lines.  Woot!

**Comment by CyrusNajmabadi** [comment]
retest vsi please

**Comment by CyrusNajmabadi** [comment]
retest vsi please

**Comment by jcouv** [comment]
Why did the second run pick up more spots that don't have comments? 

This is nice. Would it be worthwhile to make this available as a code analyzer and fix?

**Comment by Pilchie** [comment]
:-1: for now. This is likely to cause a bunch of merge pain with people working in both stabilization and master. Let's wait until we are done with stabilization for a bit.

**Comment by CyrusNajmabadi** [comment]
> Why did the second run pick up more spots that don't have comments?  

I had Roslyn open in multiple VS sessions.  Because of this, i was unable to save all the files i'd made changes to (because they were locked while the other roslyn was reading the files).

**Comment by CyrusNajmabadi** [comment]
@Pilchie Sure.  I can wait until after stabilization.  Though i'm not sure why this would be a merge issue.  Are people making lots of changes that intersect simple properties like this?

**Comment by jaredpar** [comment]
:-1: 

This code style change was not discussed with the team.

**Comment by CyrusNajmabadi** [comment]
@jaredpar I don't know what that means.  Style around these sorts of things is not generally discussed, it just happens when people make changes to a codebase that spans over multiple language versions.  See, for example, one of heejae's latest change.  In it he changed a lot of these large style properties into the new terser style  (without anyone batting an eye).  My proposal here is to simply to it in one fell swoop so that the change is not dispersed randomly through a myriad of commits.  

IMO, the new syntax is far cleaner and better than what we had.  And this change move the LS layer to now use this new style consistently in the scenarios i outlined above.  It's unclear to me (outside of concerns around timing) for why we would not want this, or why we would prefer to be in the current state when we use styles inconsistently at the LS layer *purely* because this feature didn't exist at the point in time when parts of this code were being written.

**Comment by alrz** [comment]
@CyrusNajmabadi 

Won't this PR be resurrected somehow? Cleaning up as you do other work is distracting from the actual changes being made.

**Comment by CyrusNajmabadi** [comment]
Maybe when i have some time.
